src/manifolds/Segre.jl

```diff
@@ -0,0 +1,465 @@
+"""
+get_vector(M::Segre{𝔽,V}, p, X, ::DefaultOrthonormalBasis; kwargs...) where {𝔽,V}
+
+function get_vector_orthonormal!(M::Segre{ℝ,V}, v, p, X, ::RealNumbers; kwargs...) where {V}
+    X_ = deepcopy(X)
+    v[1] = [X_[1]]
+    X_ = X_[2:end]
+    for (i, n) in enumerate(V)
+        v[i + 1] =
+            get_vector(
+                Sphere(n - 1),
+                p[i + 1],
+                X_[1:(n - 1)],
+                DefaultOrthonormalBasis();
+                kwargs...,
+            ) / p[1][1]
+        X_ = X_[n:end]
+    end
+
+    return v
+end
+
+@doc raw"""
+    function inner(M::Segre{ℝ, V}, p, u, v,)
+
+Inner product between two tangent vectors ``u = (\nu, u_1, \dots, u_d)`` and ``v = (\xi, v_1, \dots, v_d)`` at ``p \doteq (\lambda, x_1, \dots, x_d)``. This inner product is obtained by embedding the Segre manifold in the space of tensors equipped with the Euclidean metric:
+````math
+    \langle u, v \rangle_{p} = \nu \xi + \lambda^2 (\langle u_1, v_1 \rangle_{x_1} + \dots + \langle u_d, v_d \rangle_{x_d}),
+````
+where ``\nu``, ``\xi \in T_{\lambda} ℝ^{+} = ℝ`` and ``u_i``, ``v_i \in T_{x_i} S^{n_i - 1} \subset ℝ^{n_i}``.
+"""
+function inner(M::Segre{ℝ,V}, p, u, v) where {V}
+    return u[1][1] * v[1][1] + p[1][1]^2 * dot(u[2:end], v[2:end])
+end
+
+@doc raw"""
+    function rand(M::Segre{ℝ, V}; vector_at=nothing)
+
+If `vector_at` is `nothing`, return a random point on
+````math
+    ℝ^{+} \times S^{n_1 - 1} \times \dots \times S^{n_d - 1}
+````
+from a log-normal distribution on ℝ^{+} and a uniform distribution on ``S^{n_1 - 1} \times \dots \times S^{n_d - 1}``.
+
+If `vector_at` is not `nothing`, return a random tangent vector from a normal distribution on the tangent space.
+"""
+function rand(M::Segre{ℝ,V}; vector_at=nothing, kwargs...) where {V}
+    if isnothing(vector_at)
+        return [
+            rand(PositiveArrays(1); kwargs...),
+            [rand(Sphere(n - 1); kwargs...) for n in V]...,
+        ]
+    else
+        return [
+            rand(PositiveArrays(1); vector_at=vector_at[1], kwargs...),
+            [
+                rand(Sphere(n - 1); vector_at=xdot, kwargs...) for
+                (xdot, n) in zip(vector_at[2:end], V)
+            ]...,
+        ]
+    end
+end
+
+@doc raw"""
+    function get_embedding(M::Segre{𝔽,V})
+
+``\mathcal{S}`` is embedded in ``𝔽^{n_1 \times \dots \times n_d}``.
+"""
+function get_embedding(M::Segre{𝔽,V}) where {𝔽,V}
+    return Euclidean(prod(V))
+end
+
+@doc raw"""
+    function embed!(M::Segre{𝔽, V}, q, p)
+
+Embed ``p \doteq (\lambda, x_1, \dots, x_d)`` in ``𝔽^{n_1 \times \dots \times n_d}`` using the Krönecker product:
+````math
+    (\lambda, x_1, \dots, x_d) \mapsto \lambda x_1 \otimes \dots \otimes x_d.
+````
+"""
+function embed!(M::Segre{𝔽,V}, q, p) where {𝔽,V}
+    return q = kron(p...)
+end
+
+# ManifoldsBase doesn't export embed_vector! right now
+# @doc raw"""
```

Comment:
Why do you think `ManifoldsBase` should export that? It does not even exist. The function to embed tangent vectors is `embed(M, p, X)` (mutating: `embed!(M, Y, p, X)`. So we can indeed introduce this in exactly that signature.

Reply:
I fixed this and did as described here.